app/src/main/java/org/connectbot/transport/Telnet.java

```diff
@@ -68,7 +68,7 @@ public class Telnet extends AbsTransport {
 
 	static final Pattern hostmask;
 	static {
-		hostmask = Pattern.compile("^([0-9a-z.-]+)(:(\\d+))?$", Pattern.CASE_INSENSITIVE);
+		hostmask = Pattern.compile("^([0-9a-z.-_]+)(:(\\d+))?$", Pattern.CASE_INSENSITIVE);
```

Comment:
The hyphen must be the last in the regex or it changes the meaning to "all characters starting with . (0x2e) and extending to _ (0x5f)".

Reply:
Done.